src/plots/cartesian/dragbox.js

```diff
@@ -473,43 +567,67 @@ module.exports = function dragBox(gd, plotinfo, x, y, w, h, ns, ew) {
 
         var doubleClickConfig = gd._context.doubleClick,
             axList = (xActive ? xa : []).concat(yActive ? ya : []),
+            linkedAxList = (xActive || isSubplotConstrained ? xaLinked : [])
+                .concat(yActive ? yaLinked : []),
             attrs = {};
 
+        if(isSubplotConstrained) {
+            if(!xActive) linkedAxList = linkedAxList.concat(xa);
+            else if(!yActive) linkedAxList = linkedAxList.concat(ya);
+        }
+
         var ax, i, rangeInitial;
 
-        if(doubleClickConfig === 'autosize') {
+        // For reset+autosize mode:
+        // If *any* of the main axes is not at its initial range
+        // (or autoranged, if we have no initial range, to match the logic in
+        // doubleClickConfig === 'reset' below), we reset.
+        // If they are *all* at their initial ranges, then we autosize.
+        if(doubleClickConfig === 'reset+autosize') {
```

Comment:
This reorg was to fix a subtle bug:
- set explicit initial ranges for both x and y,
- autoscale one of them (doubleclick the middle of the x axis, for example)
- doubleclick the plot repeatedly: it will toggle between autoscaled x, initial y, and autoscaled y, initial x (with this change it toggles between fully autoscaled and fully initial-scaled).